scrunch/mutable_dataset.py

```diff
@@ -0,0 +1,892 @@
+        new_var = self.resource.variables.create(payload)
+        # needed to update the variables collection
+        self._reload_variables()
+        # at this point we are returning a Variable instance
+        return self[new_var['body']['alias']]
+
+    def combine_multiple_response(self, variable, map, categories=None, default=None,
+                                  name='', alias='', description=''):
+        """
+        Creates a new variable in the given dataset that combines existing
+        responses into new categorized ones
+
+            map={
+                1: 1,
+                2: [2, 3, 4]
+            },
+            categories={
+                1: "online",
+                2: "notonline"
+            }
+
+        """
+        if isinstance(variable, six.string_types):
+            parent_alias = variable
+            variable = self[variable]
+        else:
+            parent_alias = variable.alias
+
+        # TODO: Implement `default` parameter in Crunch API
+        responses = responses_from_map(variable, map, categories or {}, alias,
+                                       parent_alias)
+        payload = _VARIABLE_PAYLOAD_TMPL.copy()
+        payload['body']['name'] = name
+        payload['body']['alias'] = alias
+        payload['body']['description'] = description
+        payload['body']['derivation'] = combine_responses_expr(
+            variable.resource.self, responses)
+
+        new_var = self.resource.variables.create(payload)
+        # needed to update the variables collection
+        self._reload_variables()
+        # return an instance of Variable
+        return self[new_var['body']['alias']]
+
+    def merge(self, fork_id=None, autorollback=True):
+        """
+        :param fork_id: str or int
+            can either be the fork id, name or its number as string or int
+
+        :param autorollback: bool, default=True
+            if True the original dataset is rolled back to the previous state
+            in case of a merge conflict.
+            if False the dataset and fork are beeing left 'dirty'
+        """
+        if isinstance(fork_id, int) or (
+                isinstance(fork_id, six.string_types) and
+                fork_id.isdigit()):
+            fork_id = "FORK #{} of {}".format(fork_id, self.resource.body.name)
+
+        elif fork_id is None:
+            raise ValueError('fork id, name or number missing')
+
+        fork_index = self.resource.forks.index
+
+        forks = [f for f in fork_index
+                 if fork_index[f].get('name') == fork_id or
+                 fork_index[f].get('id') == fork_id]
+        if len(forks) == 1:
+            fork_url = forks[0]
+        else:
+            raise ValueError(
+                "Couldn't find a (unique) fork. "
+                "Please try again using its id")
+
+        body = dict(
+            dataset=fork_url,
+            autorollback=autorollback
+        )
+        self.resource.actions.post(body)
```

Comment:
Maybe it is better to use .create() here, since this will be an async response, calling `.create()` will block and wait until it finishes. If we just use `.post` then this will continue and the next line in the script that called this may work assuming that `.merge()` called finished already while it is still processing.